src/lib/reasoners/arith.ml

```diff
@@ -295,7 +295,15 @@ module Shostak
         P.add p (P.mult_const coef p3), ctx
 
     (*** <begin>: partial handling of some arith/FPA operators **)
-    | Sy.Op Sy.Float (prec, exp), [mode; x] ->
+    | Sy.Op Sy.Float, [prec; exp; mode; x] ->
+      let int_of_term t =
+        try
+          E.int_view t (* ! may be negative or null *)
+        with Failure e ->
+          Printer.print_err "%s" e;
+          assert false
+      in
+      let prec = int_of_term prec and exp = int_of_term exp in
```

Comment:
Why not just:
```suggestion
      let prec = E.int_view prec and exp = E.int_view exp in
```

Reply:
I want to reply that it is not really nice to expose a raw exception failure to the user, but on the other hand we fail with `assert false` in the other case so that wouldn't be a very compelling argument. I am OK with your suggested change.